### ISSUE TITLE
Keeping the active link for child routes.

### ISSUE THREAD

**Issue by alvaro1728** [reactjs/react-router#3237]
Yet another happy user here who wishes that when I navigate to _/projects_ or when I navigate to _/projects/:projectId_, the active state would be applied to the same link.  Here's my _routes_ method, which works fine except for that annoying active link issue:

```
export default function routes() {
  return (
    <Route component={App}>
      <IndexRoute component={Home}/>
      <Route path="/" component={Home}/>
      <Route path="projects" component={Projects}/>
      <Route path="projects/new" component={NewProject}/>
      <Route path="projects/:projectId" component={EditProject} />
      <Route path="*" component={NotFound} status={404}/>
    </Route>
  );
}
```
BTW, I tried nesting the _new_ and _:projectId_ routes inside of _projects_ but it doesn't work.  The URL changes but the current page doesn't:

```
export default function routes() {
  return (
    <Route component={App}>
      <IndexRoute component={Home}/>
      <Route path="/" component={Home}/>
      <Route path="projects" component={Projects}/>
        <Route path="new" component={NewProject}/>
        <Route path=":projectId" component={EditProject} />
      </Route>
      <Route path="*" component={NotFound} status={404}/>
    </Route>
  );
}
```
I'm on version "2.0.0" of the component.

Thanks,
Alvaro

**Comment by taion** [comment]
The latter approach is the correct way to structure your routes to work with the current `isActive` logic (and in general). Please follow up on one of our support forums if you need help getting it working. The configuration you have there does not appear correct.

**Comment by ryanflorence** [comment]
Seems like your route config should look like this:

```js
    <Route path="/" component={App}>
      <IndexRoute component={Home}/>
      <Route path="projects" component={Projects}/>
        <Route path="new" component={NewProject}/>
        <Route path=":projectId" component={EditProject} />
      </Route>
      <Route path="*" component={NotFound} status={404}/>
    </Route>
```

And then use `<Link to="/"/>` to always be active or `<IndexLink to="/"/>` to only be active when the index route is active.

**Comment by alvaro1728** [comment]
Hi @ryanflorence, thanks for your helpful reply.

I changed it exactly as you suggested and I set up my Nav bar links like this:

```
        <nav className="nav-primary">
          <ul className="icon-bar">
            <li>
              <**IndexLink** to="/" activeClassName="active">
                <i className="fa fa-home fa-2x"></i>
                <span>Home</span>
              </IndexLink>
            </li>
            <li>
              <**Link** to="/projects" activeClassName="active">
                <i className="fa fa-beer fa-2x"></i>
                <span>Projects</span>
              </Link>
            </li>
          </ul>
        </nav>
```

In my Projects page, I have a link to the "New" page, like this:

```
          <Link to="/projects/new">
            <i className="fa fa-plus"></i> New Project
          </Link>
```

And for each project in my grid, I set up a link to the "Edit" page, like this:

`<Link to={`/projects/${record.id}`}>{record.name}</IndexLink>`

So here's what's happening now:

1. When I go to mysite.com, I get the Home page with the Home link selected, which is good.
2. When I click on the Projects link, I get the Projects page and the address is changed to mysite.com/projects, which is also good.
3. When I click on the New Project link (from Projects), nothing happens!  The address is changed to mysite.com/projects/new, but the Projects page stays intact. 
4. The same thing happens when I click on one of the projects, nothing happens except the address changing to mysite.com/projects/1.

I tried changing all the Links to IndexLinks and nothing changed.  I'm not sure how they're different.

Thanks again,
Alvaro
